lib/spack/docs/basic_usage.rst

```diff
@@ -689,6 +689,109 @@ structured the way you want:
     }
 
 
+^^^^^^^^^^^^^^
+``spack diff``
+^^^^^^^^^^^^^^
+
+It's often the case that you have two versions of a spec that you need to
+disambiguate. Let's say that we've installed two variants of zlib, one with
+and one without the optimize variant:
+
+.. code-block:: console
+
+   $ spack install zlib
+   $ spack install zlib -optimize
+
+When we do ``spack find`` we see the two versions.
+
+.. code-block:: console
+
+    $ spack find zlib
+    ==> 2 installed packages
+    -- linux-ubuntu20.04-skylake / gcc@9.3.0 ------------------------
+    zlib@1.2.11  zlib@1.2.11
+
+
+Let's now say that we want to uninstall zlib. We run the command, and hit a problem
+real quickly that we have two!
+
+.. code-block:: console
+
+    $ spack uninstall zlib
+    ==> Error: zlib matches multiple packages:
+
+        -- linux-ubuntu20.04-skylake / gcc@9.3.0 ------------------------
+        efzjziy zlib@1.2.11  sl7m27m zlib@1.2.11
+
+    ==> Error: You can either:
+        a) use a more specific spec, or
+        b) specify the spec by its hash (e.g. `spack uninstall /hash`), or
+        c) use `spack uninstall --all` to uninstall ALL matching specs.
+
+Oh no! This is a good use case for ``spack diff``, which can easily show us the
+"diff" or set difference between properties for two packages. Let's try it out.
+Since the only difference between these two is the hash, we provide the hash.
+
+.. code-block::console
+
+    $ spack diff zlib@1.2.11/efzjziy zlib@1.2.11/sl7m27m
```

Comment:
This may be premature as I work my way through the diffs, but would it be enough to just write the hashes, i.e.,:

```
$ spack diff efzjziy sl7m27m
```

Reply:
Yes that works!! This is definitely a lot easier (and I'll update the docs). Sorry for my naivete - didn't even know I could do that :P

Reply:
I'll also update the json export to use the full name of the spec with the hash, so if a user saves a json export and comes back to it later they don't need to know what `/s17m27m` means.